src/scf/mixing.jl

```diff
@@ -77,11 +77,86 @@ function mix(mixing::RestaMixing, basis, ρin::RealFourierArray, ρout::RealFour
 
     ρin = ρin.fourier
     ρout = ρout.fourier
-    C = 1 - εr
+    C0 = 1 - εr
     Gsq = [sum(abs2, basis.model.recip_lattice * G) for G in G_vectors(basis)]
 
-    ρnext = @. ρin + T(mixing.α) * (ρout - ρin) * (kF^2 - C * Gsq) / (εr * kF^2 - C * Gsq)
+    ρnext = @. ρin + T(mixing.α) * (ρout - ρin) * (kF^2 - C0 * Gsq) / (εr * kF^2 - C0 * Gsq)
     # take the correct DC component from ρout; otherwise the DC component never gets updated
     ρnext[1] = ρout[1]
     from_fourier(basis, ρnext)
 end
+
+
+@doc raw"""
+The model for the susceptibility is
+```math
+\chi_0(r, r') = (-LDOS(εF, r) + LDOS(εF, r) LDOS(εF, r') / DOS(εF))
+              + \sqrt{L(x)} IFFT \frac{C_0 G^2}{4π (1 - C_0 G^2 / k_F^2)} FFT \sqrt{L(x)}
+```
+where ``C_0 = 1 - ε_r`` and the same convention for parameters is used as in `RestaMixing`.
+Additionally there is the real-space localisation function `L(r)`.
+"""
+struct HybridMixing
+    α::Real
+    εr::Real
+    kF::Real
+    localisation::Function  # Function L(r), `r` is in fractional real-space coordinates
+    rpa::Bool               # Use RPA, i.e. only apply the Hartree and not the XC Kernel
+    verbose::Bool           # Run the GMRES verbosely
+end
+function HybridMixing(;α=0.8, kF=0.8, εr=10, localisation=identity, rpa=true, verbose=false)
+    HybridMixing(α, εr, kF, localisation, rpa, verbose)
+end
+
+function mix(mixing::HybridMixing, basis, ρin::RealFourierArray, ρout::RealFourierArray;
+             εF, eigenvalues, ψ, kwargs...)
+    T = eltype(basis)
+    εr = T(mixing.εr)
+    kF = T(mixing.kF)
+    C0 = 1 - εr
+    Gsq = [sum(abs2, basis.model.recip_lattice * G) for G in G_vectors(basis)]
+
+    apply_sqrtL = identity
+    if mixing.localisation != identity
+        sqrtL = sqrt.(mixing.localisation.(r_vectors(basis)))
+        apply_sqrtL = x -> from_real(basis, sqrtL .* x.real)
+    end
+
+    # Compute the LDOS if required
+    ldos = nothing
+    if basis.model.temperature > 0
+        ldos = LDOS(εF, basis, eigenvalues, ψ)
+    end
+
+    # Solve J Δρ = ΔF with J = (1 - χ0 vc) and χ_0 given as in the docstring of the class
+    ΔF = ρout.real - ρin.real
+    devec(x) = reshape(x, size(ρin))
+    function Jop(x)
+        δρ = devec(x)
+        Jδρ = copy(δρ)
+
+        # Apply Kernel (just vc for RPA and (vc + fxc) if not RPA)
+        δV = apply_kernel(basis, from_real(basis, δρ), ρ=ρin, rpa=mixing.rpa)
+
+        # Apply Resta term of χ0
+        loc_δV = apply_sqrtL(δV).fourier
+        resta_loc_δV =  @. C0 * kF^2 * Gsq / 4T(π) / (kF^2 - C0 * Gsq) * loc_δV
+        Jδρ .-= apply_sqrtL(from_fourier(basis, resta_loc_δV)).real
+
+        # Apply LDOS term of χ0
+        if ldos !== nothing
+            Jδρ .-= (-ldos .* δV.real .+ sum(ldos .* δV.real) .* ldos ./ sum(ldos))
+        end
+
+        # Poor man's zero DC component before return
+        vec(Jδρ .-= sum(Jδρ) / length(Jδρ))
```

Comment:
It's not poor man, it's correct

Reply:
But if we do it here we probably should also do it to the input one, let's discuss